phoenix-core/src/main/java/org/apache/phoenix/expression/LikeExpression.java

```diff
@@ -289,11 +289,16 @@ public boolean evaluate(Tuple tuple, ImmutableBytesWritable ptr) {
             return true;
         }
 
-        String value = (String) PVarchar.INSTANCE.toObject(ptr, getStrExpression().getSortOrder());
-        boolean matched = pattern.matcher(value).matches();
+        String value = null;
+        if (logger.isTraceEnabled()) {
+            value = (String) PVarchar.INSTANCE.toObject(ptr, getStrExpression().getSortOrder());
+        }
+        PVarchar.INSTANCE.coerceBytes(ptr, PVarchar.INSTANCE, getStrExpression().getSortOrder(),
```

Comment:
Minor, but add a local variables at the beginning for strExpression, strSortOrder, strDataType, patternExpression, as they're repeated quote often. Also, here, change to:

    strDataType.coerceBytes(ptr, strDataType, strSortOrder);